source/unix/openssl_platform_init.c

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/allocator.h>
+
+#include <dlfcn.h>
+
+#include <aws/cal/private/opensslcrypto_common.h>
+
+static struct openssl_hmac_ctx_1_0_table hmac_ctx_1_0_table;
+static struct openssl_hmac_ctx_1_1_table hmac_ctx_1_1_table;
+static struct openssl_evp_md_ctx_table evp_md_ctx_table;
+
+struct openssl_hmac_ctx_1_0_table *g_aws_openssl_hmac_ctx_1_0_table = NULL;
+struct openssl_hmac_ctx_1_1_table *g_aws_openssl_hmac_ctx_1_1_table = NULL;
+struct openssl_evp_md_ctx_table *g_aws_openssl_evp_md_ctx_table = NULL;
+
+void aws_cal_platform_init(struct aws_allocator *allocator) {
+    (void)allocator;
+
+    void *this_handle = dlopen(NULL, RTLD_NOW);
```

Comment:
This is looking in the *current* DLL/EXE for the symbol.
What if we're building with `LibCrypto_SHARED_LIBRARY`? Wouldn't we want to look in `libcrypto.so` in that case?

Reply:
Well not exactly, I believe the way it's set up now, it will look in the process space, which includes everything that has been loaded up to that point.  It's not a complete test (libcrypto is still static) but the lookups succeed when building as shared, so aws-c-cal-tests  is separate from lib-aws-c-cal.so

From dlopen:

"If the object specified by filename has dependencies on other shared
       objects, then these are also automatically loaded by the dynamic
       linker using the same rules.  (This process may occur recursively, if
       those objects in turn have dependencies, and so on.)"

I'm assuming the same is true when using NULL to go to the main program.

From dlsym's man:

" If the symbol is not found, in the specified library or any of the libraries that were automatically loaded by dlopen() when that library was loaded, dlsym() returns NULL. (The search performed by dlsym() is breadth first through the dependency tree of these libraries.)"

Reply:
Bret is correct, this looks in the process space, including all loaded dynamic libs.